test/support/repository.go

```diff
@@ -0,0 +1,58 @@
+package support
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+)
+
+const (
```

Comment:
This took me a while to follow - but I've learned something new :-) So thanks.
Anyway, to improve readability I'd propose something like
```
const (
    groupImage = 2
    expectedRegexpResultLength = 4
)
```
and used those two constants, as groupRegistry and groupTag are not used.

Reply:
The reason I chose this approach is that I prefer naming the capture groups (groupRegistry, groupImage, groupTag) since it improves clarity and serves as inline documentation, making it easier to understand the meaning of each group when reading the code. This can be especially helpful for future maintainers.

Unfortunately, Go’s regexp package doesn’t support accessing matches by name like some other languages, so I went with the enumeration method to simulate this. While we aren’t using all the named groups in this snippet, I think keeping them helps maintain consistency and makes the intent of the regex clearer. It also leaves room for future use in case additional logic requires accessing other groups.

Reply:
I understood the reasoning at the end. I see this useful directly in regexp definition, it really servers as documentation. Constants were not clear at the beginning. But I am not against to keep it the way it is. Maybe it is a good practise.